src/NeoSharp.Core/Helpers/Guard.cs

```diff
@@ -4,12 +4,14 @@ namespace NeoSharp.Core.Helpers
 {
     public static class Guard
     {
-        public static void IsNotNull(object value, string argumentName)
+        public static T ThrowIfNull<T>(T value, string argumentName)
         {
             if (value == null)
             {
                 throw new ArgumentNullException(argumentName);
             }
+
+            return value;
```

Comment:
```csharp
return value ?? throw new ArgumentNullException(argumentName);
```

Reply:
did that .. in a different way.. do you really want with this sintaxe? why??

Reply:
try and cannot have this code
![image](https://user-images.githubusercontent.com/1231687/42053892-1892418c-7b13-11e8-82aa-2512ee06b7ba.png)